app/models/comment_notifier.rb

```diff
@@ -22,8 +22,10 @@ def send_reply_email
   end
 
   def email_on_comment?
-    commentable_author = @comment.commentable.author
-    commentable_author != @author && commentable_author.email_on_comment?
+    unless @comment.commentable.is_a?(Poll)
```

Comment:
This is fine... maybe it's a little clearer this way:
`return false if @comment.commentable.is_a?(Poll)`